### ISSUE TITLE
[P3][dogfood] Kotlin: primary constructors report return_type "void" (Java-ism) instead of null/Unit — hard-coded at kotlin_helpers.py:450

### ISSUE THREAD

**Issue by aimasteracc** [aimasteracc/tree-sitter-analyzer#761]
**Severity:** P3  |  **Surface:** CLI --advanced --table json (Kotlin extraction, methods[] entries with is_constructor=true)
_Found by parallel dogfood audit round 2 on develop @ 471ac5f5; adversarially re-reproduced + dup-checked by an independent verifier._

### Observed
All four constructors print return_type='void': User void / Success void / Error void / UserManager void. Meanwhile real Kotlin functions correctly use Unit ('main Unit') or actual types ('display String'). The 'void' literal is hard-coded at tree_sitter_analyzer/languages/kotlin_helpers.py:450 (return_type="void") in the primary-constructor builder, while the function builder at line 307 defaults to 'Unit'.

### Expected
Kotlin constructors should report return_type as null/omitted (constructors have no return type) — or at minimum the Kotlin-correct value, never the Java placeholder 'void'. The intra-plugin inconsistency (constructors='void' vs functions='Unit') confirms a hard-coded Java default leaking in. Note: any fix that re-pins the extractor version must follow the exact-assertion test rule and bump the extractor version markers.

### Repro
```bash
cd /Users/aisheng.yu/git-private/tsa-dogfood && uv run python -m tree_sitter_analyzer examples/Sample.kt --advanced --table json 2>/dev/null | python3 -c "import json,sys; d=json.load(sys.stdin); [print(m['name'],m['return_type']) for m in d['methods'] if m.get('is_constructor')]"
```

### Verifier notes
Re-ran the exact repro in /Users/aisheng.yu/git-private/tsa-dogfood on develop. All four Kotlin constructors (User, Success, Error, UserManager) report return_type='void', while real functions correctly report 'Unit' (e.g. main -> 'Unit') or actual types (display -> 'String'). Root cause located: tree_sitter_analyzer/languages/kotlin_helpers.py:450 hard-codes return_type='void' in the primary-constructor Function(...) builder, whereas the function path (line 307) defaults to 'Unit'. This is a Java-ism leaking into the Kotlin plugin — Kotlin has no 'void' type. Not a flag/user error (--advanced --table json is the documented extraction path). Dedup: gh issue search for 'Kotlin', 'void return_type constructor', and 'extraction' returns no Kotlin-specific extraction issue; round-1 issues #745-#753 cover JS/Go/C/C++ (different languages/symptoms), so this is not a duplicate.

**Comment by aimasteracc** [comment]
Fixed in a recent merge to develop. Verified as resolved.
